admin/server/deployment.go

```diff
@@ -150,3 +152,79 @@ func (s *Server) TriggerRedeploy(ctx context.Context, req *adminv1.TriggerRedepl
 
 	return &adminv1.TriggerRedeployResponse{}, nil
 }
+
+// GetDeploymentCredentials returns runtime info and JWT on behalf of a specific user, or alternatively for a raw set of JWT attributes
+func (s *Server) GetDeploymentCredentials(ctx context.Context, req *adminv1.GetDeploymentCredentialsRequest) (*adminv1.GetDeploymentCredentialsResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.organization", req.Organization),
+		attribute.String("args.project", req.Project),
+		attribute.String("args.branch", req.Branch),
+	)
+
+	proj, err := s.admin.DB.FindProjectByName(ctx, req.Organization, req.Project)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if proj.ProdDeploymentID == nil {
+		return nil, status.Error(codes.InvalidArgument, "project does not have a deployment")
+	}
+
+	prodDepl, err := s.admin.DB.FindDeployment(ctx, *proj.ProdDeploymentID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if req.Branch != "" && req.Branch != prodDepl.Branch {
+		return nil, status.Error(codes.InvalidArgument, "project does not have a deployment for given branch")
+	}
+
+	claims := auth.GetClaims(ctx)
+	permissions := claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID)
+
+	// If the user is not a superuser, they must have ManageProd permissions
+	if !permissions.ManageProd && !claims.Superuser(ctx) {
+		return nil, status.Error(codes.PermissionDenied, "does not have permission to manage deployment")
+	}
+
+	var attr map[string]any
+	switch id := req.For.(type) {
+	case *adminv1.GetDeploymentCredentialsRequest_UserId:
+		attr, err = s.jwtAttributesForUser(ctx, permissions, claims.OwnerID(), proj.OrganizationID)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+	case *adminv1.GetDeploymentCredentialsRequest_Attrs:
+		attr = id.Attrs.AsMap()
+	default:
+		return nil, status.Error(codes.InvalidArgument, "id must be either user or attrs")
+	}
```

Comment:
1. Call the variable `for` instead of `id` to match the `req.For` name
2. Should pass `for.UserId` instead of `claims.OwnerID()` to `jwtAttributesForUser`
3. In `default` case, just return `invalid 'for' type`